### DISCUSSION TITLE
Assert.InRange and IComparable<T>

### DISCUSSION THREAD

**Discussion by StefanBertels** [xunit/xunit#2176] [Ideas]
Hi, seems that 

`Assert.InRange(T expected, T min, T max)`

only works for types implementing non-generic `IComparable`. 
I have types implementing `IComparable<T>` only and they cannot be used with above `InRange`.

How about adding support for (type-safe) `IComparable<T>` interface?

I know there are workarounds but I hope there is a straight-forward way to make the simple `InRange` function usable.

**Comment by bradwilson** [comment]
Sounds good to me! Feel free to open an issue and I'll put it on the 3.0 backlog.

**Reply by bradwilson** [reply]
Going to mark this as answered. Please let me know if you have any further questions.

**Reply by StefanBertels** [reply]
Thanks for looking at this and your answers. I _solved_ my issue meanwhile by getting `IComparable` (non-generic) implemented upstream and it looks like implementing both interfaces is a defacto standard (e.g. dotnet core runtime).
